From 777afbdcedba732c6a2d54ccfdff08f22eaf1403 Mon Sep 17 00:00:00 2001
From: Silvio <silvio.traversaro@iit.it>
Date: Tue, 20 Aug 2024 19:13:15 +0200
Subject: [PATCH] Install sdk_info.h even if RERUN_INSTALL_RERUN_C option is
 OFF

---
 rerun_cpp/CMakeLists.txt | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/rerun_cpp/CMakeLists.txt b/rerun_cpp/CMakeLists.txt
index fc4c8160588f..f20b52ed4975 100644
--- a/rerun_cpp/CMakeLists.txt
+++ b/rerun_cpp/CMakeLists.txt
@@ -153,8 +153,12 @@ install(TARGETS rerun_sdk
     INCLUDES DESTINATION include
 )
 
+# Some public headers are shared by both the C++ and C sdk, so we install
+# them with the C++ headers also if RERUN_INSTALL_RERUN_C is OFF
+set(RERUN_PATTERN_HEADER_SHARED_BETWEEN_CXX_AND_C "sdk_info.h")
+
 # Add all C++ headers to the install.
-install(DIRECTORY "${RERUN_CPP_SOURCE_DIR}/" TYPE INCLUDE FILES_MATCHING PATTERN "*.hpp")
+install(DIRECTORY "${RERUN_CPP_SOURCE_DIR}/" TYPE INCLUDE FILES_MATCHING PATTERN "*.hpp" PATTERN ${RERUN_PATTERN_HEADER_SHARED_BETWEEN_CXX_AND_C})
 
 option(RERUN_INSTALL_RERUN_C "Install rerun_c file." ON)
 
@@ -168,7 +172,7 @@ endif()
 if(RERUN_INSTALL_RERUN_C)
     # CMake doesn't allow installing imported targets which is why we need to add this as a file.
     get_target_property(RERUN_C_LIB_LOCATION rerun_c IMPORTED_LOCATION)
-    install(DIRECTORY "${RERUN_CPP_SOURCE_DIR}/" TYPE INCLUDE FILES_MATCHING PATTERN "*.h")
+    install(DIRECTORY "${RERUN_CPP_SOURCE_DIR}/" TYPE INCLUDE FILES_MATCHING PATTERN "*.h" PATTERN ${RERUN_PATTERN_HEADER_SHARED_BETWEEN_CXX_AND_C} EXCLUDE)
     install(FILES ${RERUN_C_LIB_LOCATION} DESTINATION lib)
 endif()

